### PR TITLE
Add unitTest for `Tensor==constant` for ifElse in dygraph2static 

### PR DESCRIPTION
**Pull request by Aurelius84**
+ Add `generic_visit` in `gast.Call`
+ Add unitTest for `Tensor==constant` for ifElse in dygraph2static

`Tensor == constant` is support in dygraph, see detail in https://github.com/PaddlePaddle/Paddle/pull/22892

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_ifelse.py [hunk 1]
```diff
@@ -116,6 +116,30 @@ def setUp(self):
         self.dyfunc = dyfunc_ifExp_with_while
 
 
+def dyfunc_ifExp_with_while2(x):
```
Comment (zhhsplendid) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_ifelse.py:119]: Why is it named `dyfunc_ifExp_with_while2`? I didn't see `while` in this test

Comment (Aurelius84) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_ifelse.py:119]: I forgot to rename it and will fix it in https://github.com/PaddlePaddle/Paddle/pull/23478

Comment (zhhsplendid) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_ifelse.py:119]: After offline discussion, we will merge this at first due to CI time, then rename to `dyfunc_ifExp` immediately at https://github.com/PaddlePaddle/Paddle/pull/23478

**Review by zhhsplendid — APPROVED** [review]
Approved

**Pull request merged**